frontend/test/metabase/scenarios/admin/databases/add.cy.spec.js

```diff
@@ -0,0 +1,122 @@
+import { signInAsAdmin, snapshot, restore } from "__support__/cypress";
+import { saveFile } from "jsrsasign";
+
+function typeField(name, value) {
+  cy.get(`input[name="${name}"]`)
+    .clear()
+    .type(value)
+    .blur();
+}
+
+function toggleFieldWithDisplayName(displayName) {
+  cy.contains(displayName)
+    .closest(".Form-field")
+    .find("a")
+    .click();
+}
+
+describe("admin > databases > add", () => {
+  before(snapshot);
+  after(restore);
+
+  beforeEach(() => {
+    signInAsAdmin();
+    cy.server();
+  });
+
+  it("should work and shouldn't let you accidentally add db twice", () => {
+    cy.route({
+      method: "POST",
+      url: "/api/database",
+      response: { id: 42 },
+      delay: 1000,
+    }).as("createDatabase");
+
+    cy.visit("/admin/databases/create");
+
+    typeField("name", "Test db name");
+    typeField("dbname", "test_postgres_db");
+    typeField("user", "uberadmin");
+
+    cy.contains("button", "Save")
+      .should("not.be.disabled")
+      .click();
+    cy.contains("button", "Saving...").should("be.disabled");
```

Comment:
I think this is technically a race, but it probably won't ever matter. If the save completes before Cypress processes this `contains`, the test will fail.

It doesn't matter as networks calls will probably always lose that race, but it made me wonder if there's a more robust way to write the test. Maybe you'd need to stub the request to ensure that it won't complete too early?

Reply:
I assume Cypress immediately begins looking for "Saving..." after performing the click. I guess the question is how often it checks. If it's polling at a rate slower than the fastest network request, that could be an issue. If it's listening for DOM mutations it should be fine.